documentation/dt-bindings.md

```diff
@@ -0,0 +1,55 @@
+## Secure World OP-TEE DT bindings
+
+OP-TEE has a formal DT binding at
+
+```
+	firmware {
+		optee {
+		};
+	};
+```
+
+The Linux kernel device tree bindings documentation
+explains the properties that may appear there that
+are of interest to Normal World.
+
+OP-TEE may also be passed a DTB at boot-time, which
+may be different to, or modified from, the DTB passed
+to Normal World.
+
+The following properties inside /firmware/optee are
+understood by OP-TEE itself in the Secure World.
+
+### secure-device-id
+
+This allows the bootloader to declare an array of bytes
+which OP-TEE will use as the Secure Device ID.  For
+example OP-TEE uses this information to produce the
+key used for Secure Storage on that device.
+
+Since almost no SoC has public documentation on how to
+read the real device key data from e-fuses, this allows
+alternate ways to get per-device stable identity bits,
+eg, read the eMMC CID serial number on the same board.
+
+This may produce less data than a real key on the SoC,
+but it is still very useful since it requires no
+configuration to get a unique number.
```

Comment:
Mmm that is not what is going on with this, that is also why it was called a "pseudo device ID" in the first patches.

This isn't trying to "be" the device key e-fuses, it's trying to solve the existing situation that for OP-TEE there is zero logical difference between any number of same boards like Hikey.

That directly leads to really undesirable situations in OP-TEE itself like Secure Storage is not secure at all.  And real applications in the TA also need this, for example tracking of unprovisioned devices to get the assigned key data in the factory, or to recover them if they need it in the field.

While this number belongs to the eMMC and not the SoC, and while it is only 32-bits, in every other way it is real, usable device-specific key data that needs no configuration.

> not much more secure than hard-coding a security key in the clear text binary

This data comes out of the board at runtime, so it doesn't need unique TA binaries per device.

Yes it's not secure in itself at all... it's not fighting e-fuses it's fighting every board is 100% the same to OP-TEE when OP-TEE's business involves per-device crypto and defeating copying secure storage content between individual devices.

I will send another documentation patch in the next weeks addressing this big missing section at the start of the docs listing the steps you have to take to make OP-TEE secure, we can mention this there.